cmdr/src/giti/branch/checkout.rs

```diff
@@ -0,0 +1,149 @@
+/*
+ *   Copyright (c) 2023 R3BL LLC
+ *   All rights reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+use std::process::Command;
+
+use r3bl_ansi_color::{AnsiStyledText, Color, Style};
+use r3bl_rs_utils_core::{log_debug, CommonResult};
+use r3bl_tuify::{select_from_list_with_multi_line_header,
+                 SelectionMode,
+                 StyleSheet,
+                 LIGHT_GRAY_COLOR,
+                 SUCCESS_COLOR};
+
+use super::{get_branches, try_get_current_branch};
+use crate::giti::{report_unknown_error_and_propagate, single_select_instruction_header};
+
+pub fn try_checkout_branch() -> CommonResult<()> {
+    let default_header_style = [
+        Style::Foreground(Color::Rgb(171, 204, 242)),
+        Style::Background(Color::Rgb(31, 36, 46)),
+    ];
+
+    let instructions_and_branches = {
+        let mut instructions_and_branches = single_select_instruction_header();
+        let header = AnsiStyledText {
+            text: "Select a branch to switch to:",
+            style: &default_header_style,
+        };
+        instructions_and_branches.push(vec![header]);
+        instructions_and_branches
+    };
+
+    let current_branch = try_get_current_branch()?;
+
+    if let Ok(branches) = get_branches() {
+        // Ask user to select a branch to check out to.
+        let maybe_selected_branch = select_from_list_with_multi_line_header(
+            instructions_and_branches,
+            branches,
+            Some(20),
+            None,
+            SelectionMode::Single,
+            StyleSheet::default(),
+        );
+
+        // If user selected a branch, then check out to it.FF
+        if let Some(selected_branch) = maybe_selected_branch {
+            let selected_branch = &selected_branch[0];
+            let selected_branch = selected_branch.trim_start_matches("(current) ");
+            log_debug(format!("selected_branch -------> `{selected_branch}`"));
+            log_debug(format!("current_branch ------> `{current_branch}`"));
```

Comment:
If you intend to keep them, best to wrap them in a `call_if_true!(flag, {..})` where flag is some constant that you might want to put in `lib.rs` or `mod.rs` file.

Reply:
Thanks! Going to delete them for now. They are not that useful and can be always added back.